config/opal_setup_cc.m4

```diff
@@ -23,6 +23,115 @@ dnl
+          [opal_proc_cc_c11_helper_tmp=0])
+
+    AC_DEFINE_UNQUOTED([OPAL_C_HAVE_C11_ATOMIC], [$opal_proc_cc_c11_helper_tmp], [Whether C compiler supports C11 atomics])
+
+    AS_IF([test "x$opal_prog_cc_c11_helper__atomic_available" = "xyes"], [opal_proc_cc_c11_helper_tmp=1],
+          [opal_proc_cc_c11_helper_tmp=0])
+
+    AC_DEFINE_UNQUOTED([OPAL_C_HAVE__ATOMIC], [$opal_proc_cc_c11_helper_tmp], [Whether C compiler supports __Atomic])
+
+    dnl At this time Open MPI only requires thread local and the atomic convenience types
+    AS_IF([test "x$opal_prog_cc_c11_helper__thread_local_available" = "xyes" && test "x$opal_prog_cc_c11_helper_atomic_var_available" = "xyes"],
+          [$2], [$3])
+
+    CFLAGS=$opal_prog_cc_c11_helper_CFLAGS_save
+
+    OPAL_VAR_SCOPE_POP
+])
+
+AC_DEFUN([OPAL_PROG_CC_C11],[
+    OPAL_VAR_SCOPE_PUSH([opal_prog_cc_c11_flags])
+    if test -z "$opal_cv_c11_supported" ; then
+        opal_cv_c11_supported=no
+        opal_cv_c11_flag_required=yes
+
+        AC_MSG_CHECKING([if $CC requires a flag for C11])
+
+        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+#if __STDC_VERSION__ < 201112L
+#error "Need a flag for C11"
+#endif
+                                           ]],[])],
+                          [opal_cv_c11_flag_required=no])
+
+        AC_MSG_RESULT([$opal_cv_c11_flag_required])
+
+        if test "x$opal_cv_c11_flag_required" = "xno" ; then
+            AC_MSG_NOTICE([verifying $CC supports C11 without a flag])
+            OPAL_PROG_CC_C11_HELPER([], [], [opal_cv_c11_flag_required=yes])
+        fi
+
+        if test "x$opal_cv_c11_flag_required" = "xyes" ; then
+            opal_prog_cc_c11_flags="-std=gnu11 -std=c11 -c11"
+
+            AC_MSG_NOTICE([checking if $CC supports C11 with a flag])
+            opal_cv_c11_flag=
+            for flag in $(echo $opal_prog_cc_c11_flags | tr ' ' '\n') ; do
+                OPAL_PROG_CC_C11_HELPER([$flag],[opal_cv_c11_flag=$flag],[])
+                if test "x$opal_cv_c11_flag" != "x" ; then
+                    CFLAGS="$CFLAGS $opal_cv_c11_flag"
+                    AC_MSG_NOTICE([using $flag to enable C11 support])
+                    opal_cv_c11_supported=yes
+                    break
+                fi
+            done
+        else
+            AC_MSG_NOTICE([no flag required for C11 support])
+            opal_cv_c11_supported=yes
+        fi
+
+        if test $opal_cv_c11_supported = no ; then
+            AC_MSG_WARN([Open MPI recommends a C11 compiler])
+            # uncomment the following line and update the warning when
+            # a C11 compiler is required
+            # AC_MSG_ERROR([Aborting.])
+            # From Open MPI 1.7 on we require a C99 compiant compiler
+            AC_PROG_CC_C99
+            # The result of AC_PROG_CC_C99 is stored in ac_cv_prog_cc_c99
+            if test "x$ac_cv_prog_cc_c99" = xno ; then
+                AC_MSG_WARN([Open MPI requires a C99 compiler])
```

Comment:
This warning message is now no longer strictly accurate -- perhaps it should become:

> Open MPI requires a C compiler that supports C99 or later